### PR TITLE
feat: TFOD Hyperparameters parsing

### PR DESCRIPTION
**Pull request by javoweb**

### CODE REVIEW

File: workflows/tf-object-detection-training/utils.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        pipeline_config.model.faster_rcnn.second_stage_post_processing.batch_non_max_suppression.score_threshold = float(model_params['second_stage_nms_score_threshold'])
+        pipeline_config.model.faster_rcnn.second_stage_post_processing.batch_non_max_suppression.iou_threshold = float(model_params['second_stage_nms_iou_threshold'])
+        pipeline_config.model.faster_rcnn.second_stage_post_processing.batch_non_max_suppression.max_detections_per_class = int(model_params['second_stage_max_detections_per_class'])
+        pipeline_config.model.faster_rcnn.second_stage_post_processing.batch_non_max_suppression.max_total_detections = int(model_params['second_stage_max_detections_max_total_detections'])
+        pipeline_config.model.faster_rcnn.second_stage_localization_loss_weight = float(model_params['second_stage_localization_loss_weight'])
+        pipeline_config.model.faster_rcnn.second_stage_classification_loss_weight = float(model_params['second_stage_classification_loss_weight'])
+
+        pipeline_config.train_config.optimizer.momentum_optimizer.learning_rate.manual_step_learning_rate.initial_learning_rate = float(model_params['initial-learning-rate'])
```
Comment (rushtehrani) [inline workflows/tf-object-detection-training/utils.py:139]: Can we use `_` for these as well but also provide backwards compatibility for previous Workflows?

Comment (javoweb) [reply workflows/tf-object-detection-training/utils.py]: Done! Added legacy compatibility too. Supported "initial-learning-rate" and "learning_rate" too as was in docs.

File: workflows/tf-object-detection-training/main.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def main(params):
         '/mnt/src/tf/research/object_detection/legacy/train.py',
         '--train_dir=/mnt/output/',
         '--pipeline_config_path=/mnt/output/pipeline.config',
-        '--num_clones={}'.format(params['num-clones'])
+        '--num_clones={}'.format(params['num_clones'])
```
Comment (rushtehrani) [inline workflows/tf-object-detection-training/main.py:46]: @javoweb we'd need to also support the old keys (with the `-`) for the existing params to retain backwards compatibility.

Comment (javoweb) [reply workflows/tf-object-detection-training/main.py:46]: Yes, I'm doing it on a new function  `legacy_params_compatibility` I'm parsing old keys to the newer ones. But the rest of the code use the new keys with the `_` only.

Comment (rushtehrani) [reply workflows/tf-object-detection-training/main.py:46]: Ah, missed that. Looks good!

File: workflows/tf-object-detection-training/utils.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+def get_default_params(model):
+    with open('defaults.json', 'r') as f:
+        default_dict = json.load(f)
+    return default_dict[model].copy()
+
+def legacy_params_compatibility(params):
+    new_params = params.copy()
+    if "num-clones" in params:
+        new_params["num_clones"]=params["num-clones"]
```
Comment (rushtehrani) [inline workflows/tf-object-detection-training/utils.py:19]: can you add spaces before and after `=`?

**Comment by rushtehrani** [comment]
I'm going to do one more test on this and merge it.

File: workflows/tf-object-detection-training/utils.py (hunk 3 above)
Comment (javoweb) [reply workflows/tf-object-detection-training/utils.py]: Done.

**Pull request merged**